### PR TITLE
Protein bar Fixes and Tweaks

### PR DESCRIPTION
**Pull request by chesse20**
## About The Pull Request
Changes the way protein bars are generated so that they are no longer creating and then deleting a fake protein bar, just to create a real protein bar. As a side effect of this you will no longer be able to purchase specific flavors of protein bars.

## Why It's Good For The Game
Because 
1. the protein bars did not vend into your hands like old protein bars
2. the vendors vended fake protein bars into your hands that you could not see
3. the fake protein bars would prevent you from picking anything up even though they were deleted as soon as they entered your hands

## Changelog
:cl:
del: You can no longer vend specific flavors of protein bars
fix: Stopped filling your hands with fake protein bars
fix: nutriment has taste as intended
qol: Protein bars now vend into your hands as they should
/:cl:

### CODE REVIEW

File: code/game/objects/items/marine_gear.dm [hunk 1]
```diff
@@ -53,82 +51,31 @@
...
+		list("blueberry TGMC protein bar","A nice blueberry crunch into your otherwise stale and boring protein bar.","#4e39c5",/datum/reagent/consumable/protein/blueberry)
+	)
+	///list of protein bar stats to change
+	var/list/picked = pick(randlist)
+	///amount of protein mix to put inside protein bars (8 to emulate old protein mec
+	var/protein_to_add = 8 //no magic numbers
+	name = picked[1]
+	desc = picked[2]
+	set_greyscale_colors(picked[3])
+	src.reagents.add_reagent(picked[4],protein_to_add)
```
Comment (BraveMole) [inline code/game/objects/items/marine_gear.dm:77]:
```suggestion
	reagents.add_reagent(picked[4],protein_to_add)
```

no src.

File: code/game/objects/items/marine_gear.dm [hunk 2]
```diff
@@ -53,82 +51,31 @@
...
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/four,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/five,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/six,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/seven,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/eight,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/nine)
-	new protein_bar(src.loc)
-	qdel(src)
+	///list of producable protein bars
+	var/list/randlist = list(
```
Comment (BraveMole) [inline code/game/objects/items/marine_gear.dm:59]: this should be a static variable on the protein

**Review by BraveMole — CHANGES_REQUESTED** [review]
You need to fix the taste as well, rather than having specific code for protein flavour make it on food/snack

File: code/game/objects/items/marine_gear.dm [hunk 3]
```diff
@@ -53,82 +51,31 @@
...
+		list("stale TGMC protein bar","The most fake looking protein bar you have ever laid eyes on, covered in the a subtitution chocolate. The powder used to make these is a subsitute of a substitute of whey substitute.","#f37d43",/datum/reagent/consumable/protein),
+		list("mint TGMC protein bar","Stale old protien bar, with an almost minty freshness to it, but not fresh enough","#61b36e",/datum/reagent/consumable/protein/mint),
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:61]:
```suggestion
		list("mint TGMC protein bar","Stale old protien bar, with an almost minty freshness to it, but not fresh enough.","#61b36e",/datum/reagent/consumable/protein/mint),
```

Comment (XSlayer300) [reply code/game/objects/items/marine_gear.dm:61]: heckin' please you haven't even fixed this too >:(

File: code/modules/reagents/reagents/food.dm [hunk 4]
```diff
@@ -102,35 +102,35 @@
 
 /datum/reagent/consumable/protein/mint
 	name = "Mint Protein Mix"
-	taste_description = "minty protein"
+	//taste_description = "minty protein"
 
 /datum/reagent/consumable/protein/grape
 	name = "Grape Protein Mix"
```
Comment (TiviPlus) [inline code/modules/reagents/reagents/food.dm:108]: Commented out code

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Also see pending bravemole comment

File: code/modules/reagents/reagents/food.dm (hunk 4 above)
Comment (chesse20) [reply code/modules/reagents/reagents/food.dm:108]: ill remove it once i fix flavor still workin

**Comment by chesse20** [comment]
The protein bars now have flavor, don't use seperate reagents for each flavor, and the flavor now works thanks to @BraveMole tutoring me on how to build and debug byond code from vsc

**Comment by chesse20** [comment]
image

File: code/game/objects/items/marine_gear.dm [hunk 5]
```diff
@@ -68,6 +65,11 @@
 		list("chicken TGMC protein bar","Protein bar covered with chicken powder one might find in ramen. Get some extra sodium with your protein.","#cccc00",list= ("powdered chicken")),
 		list("blueberry TGMC protein bar","A nice blueberry crunch into your otherwise stale and boring protein bar.","#4e39c5",list("blueberry" = 1))
 	)
+
+/obj/item/reagent_containers/food/snacks/protein_pack/Initialize()
+	. = ..()
+
+
```
Comment (BraveMole) [inline code/game/objects/items/marine_gear.dm]:
```suggestion
	. = ..()
```

**Review by BraveMole — APPROVED** [review]

File: code/game/objects/items/marine_gear.dm [hunk 6]
```diff
@@ -42,93 +42,39 @@
...
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/three,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/four,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/five,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/six,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/seven,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/eight,
-		/obj/item/reagent_containers/food/snacks/protein_pack_base/nine)
-	new protein_bar(src.loc)
-	qdel(src)
+	///list of protein bar stats to change
```
Comment (hyper2snyper) [inline code/game/objects/items/marine_gear.dm:70]: dont need to autodoc non object vars.

**Review by hyper2snyper — APPROVED** [review]
Yeah this looks good

File: code/game/objects/items/marine_gear.dm (hunk 6 above)
Comment (chesse20) [reply code/game/objects/items/marine_gear.dm:70]: sure i can remove that

**Review by BraveMole — APPROVED** [review]

**Pull request merged**